cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -246,7 +248,226 @@ EXEC strQry // SQL Injection vulnerability!
+
+DO: Use a strong encryption routine such as AES-512 where personally identifiable data needs to be restored to it's original format. Do not encrypt passwords. Protect encryption keys more than any other asset. Apply the following test: Would you be happy leaving the data on a spreadsheet on a bus for everyone to read. Assume the attacker can get direct access to your database and protect it accordingly. More information [here](Transport_Layer_Protection_Cheat_Sheet.md).
+
+DO: Use TLS 1.2 for your entire site. Get a free certificate [LetsEncrypt.org](https://letsencrypt.org/).
+
+DO NOT: [Allow SSL, this is now obsolete](https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices#22-use-secure-protocols).
+
+DO: Have a strong TLS policy (see [SSL Best Practises](http://www.ssllabs.com/projects/best-practises/)), use TLS 1.2 wherever possible. Then check the configuration using [SSL Test](https://www.ssllabs.com/ssltest/) or [TestSSL](https://testssl.sh/).
+
+DO: Ensure headers are not disclosing information about your application. See [HttpHeaders.cs](https://github.com/johnstaveley/SecurityEssentials/blob/master/SecurityEssentials/Core/HttpHeaders.cs) , [Dionach StripHeaders](https://github.com/Dionach/StripHeaders/), disable via `web.config` or [startup.cs](https://medium.com/bugbountywriteup/security-headers-1c770105940b):
+
+e.g Web.config
+
+```xml
+<system.web>
+    <httpRuntime enableVersionHeader="false"/>
+</system.web>
+<system.webServer>
+    <security>
+        <requestFiltering removeServerHeader="true" />
+    </security>
+    <httpProtocol>
+        <customHeaders>
+            <add name="X-Content-Type-Options" value="nosniff" />
+            <add name="X-Frame-Options" value="DENY" />
+            <add name="X-Permitted-Cross-Domain-Policies" value="master-only"/>
+            <add name="X-XSS-Protection" value="1; mode=block"/>
+            <remove name="X-Powered-By"/>
+        </customHeaders>
+    </httpProtocol>
+</system.webServer>    
+```
+
+e.g Startup.cs
+
+``` csharp
+app.UseHsts(hsts => hsts.MaxAge(365).IncludeSubdomains());
+app.UseXContentTypeOptions();
+app.UseReferrerPolicy(opts => opts.NoReferrer());
+app.UseXXssProtection(options => options.EnabledWithBlockMode());
+app.UseXfo(options => options.Deny());
+
+app.UseCsp(opts => opts
+ .BlockAllMixedContent()
+ .StyleSources(s => s.Self())
+ .StyleSources(s => s.UnsafeInline())
+ .FontSources(s => s.Self())
+ .FormActions(s => s.Self())
+ .FrameAncestors(s => s.Self())
+ .ImageSources(s => s.Self())
+ .ScriptSources(s => s.Self())
+ );
+```
+
+For more information about headers can be found [here](https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#xpcdp).
+
+## A4 [XML External Entities (XXE)](XML_External_Entity_Prevention_Cheat_Sheet.md#net)
+
```

Comment:
You can replace all the XXE section by a link to the XXE CS [.NET section](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.md#net) because it's already covered